docs/source/advanced/response-based-codegen.mdx

```diff
@@ -61,20 +72,19 @@ when (hero) {
   }
 }
 ```
-## Accessors
 
-As a convenience, methods named `asXyz` returning the possible implementations are also generated. They show up in the IDE's completion, and allow avoiding manual casts. 
+#### Accessors
 
-In our example they can be used like so:
+As a convenience, the `responseBased` codegen generates methods with the name pattern `as<Typename>` (e.g., `asDroid` or `asHuman`) that enable you to avoid manual casting:
```

Comment:
It's not always "Typename". For an example: 

```graphql
query GetAnimal {
  ... on Pet {
    name
  }
  ... on WarmBlooded {
    temperature
  }
}
```

That will have `asPetWarmBlooded()` to avoid generating classes for each and every concrete type. We're [trying to define proper terminology for this](https://apollographql.quip.com/d7KaAOA9f8bz/Client-Glossary) which is not easy. I'd suggest "Shape" for the time being:

```suggestion
As a convenience, the `responseBased` codegen generates methods with the name pattern `as<ShapeName>` (e.g., `asDroid` or `asHuman`) that enable you to avoid manual casting:
```